app/models/survey.rb

```diff
@@ -1,4 +1,6 @@
 class Survey < ActiveRecord::Base
+  before_create :generate_ref_code
```

Comment:
I'm working on getting a new survey to be saved, and though this was not in the model when i was getting  the "must save the parent before you can call create on a child" error, the solution i found alludes to using after_create instead of before_create, on this method, so this might be something we may want to change if we come upon that error again. 

(reference : http://stackoverflow.com/questions/19711610/getting-error-on-ruby-model-you-cannot-call-create-unless-the-parent-is-saved)